src/me/truemb/rentit/api/AdvancedChestsUtils.java

```diff
@@ -0,0 +1,90 @@
+package me.truemb.rentit.api;
+
+import com.sk89q.worldedit.math.BlockVector3;
+import me.truemb.rentit.enums.RentTypes;
+import me.truemb.rentit.main.Main;
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.World;
+import org.bukkit.block.BlockFace;
+import org.bukkit.block.data.Directional;
+import us.lynuxcraft.deadsilenceiv.advancedchests.AdvancedChestsAPI;
+import us.lynuxcraft.deadsilenceiv.advancedchests.chest.AdvancedChest;
+import us.lynuxcraft.deadsilenceiv.advancedchests.chest.ChestBuilder;
+import us.lynuxcraft.deadsilenceiv.advancedchests.utils.LocationUtils;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class AdvancedChestsUtils {
+    private final Main instance;
+
+    private final boolean isEnabled;
+
+    public AdvancedChestsUtils(Main plugin) {
+        this.instance = plugin;
+        this.isEnabled = this.instance.manageFile().getBoolean("Options.useAdvancedChests")
+                && Bukkit.getServer().getPluginManager().isPluginEnabled("AdvancedChests");
+    }
+
+    public boolean isEnabled() {
+        return this.isEnabled;
+    }
+
+    public Map<String, List<String>> getChestsInArea(BlockVector3 min, BlockVector3 max, World world) {
+        Map<String, List<String>> chests = new HashMap<>();
+
+        if (!this.isEnabled) {
+            return chests;
+        }
+
+        for(int x = min.getBlockX(); x <= max.getBlockX(); x++)
+            for(int y = min.getBlockY(); y <= max.getBlockY(); y++)
+                for(int z = min.getBlockZ(); z <= max.getBlockZ(); z++) {
+                    Location location = new Location(world, x, y, z);
+                    AdvancedChest chest = AdvancedChestsAPI.getChestManager().getAdvancedChest(location);
+
+                    if (chest != null) {
+                        /*
+                         * If there is an advanced chest in the given location, we recover the location
+                         * directional, so we can decorate our serialized data with the chest facing.
+                         */
+                        Directional d = (Directional) location.getBlock().getBlockData();
```

Comment:
Where are you doing this `location.getBlock().getState().getBlockData()`? Also, I can't get the setArea command to fail, any particular configuration of the area / elements inside?

Reply:
I replaced Line 54 since I am getting the ClassCastError (see pastebin link).
After that I am getting the second error for accessing from an async method (/<shop/hotel> setarea)